arch/posix/core/fatal.c

```diff
@@ -34,41 +37,40 @@ const NANO_ESF _default_esf = {
 FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
 		const NANO_ESF *esf)
 {
-#ifdef CONFIG_PRINTK
+	LOG_PANIC();
+
 	switch (reason) {
 	case _NANO_ERR_CPU_EXCEPTION:
 	case _NANO_ERR_SPURIOUS_INT:
 		break;
 
 	case _NANO_ERR_INVALID_TASK_EXIT:
-		printk("***** Invalid Exit Software Error! *****\n");
+		LOG_ERR("***** Invalid Exit Software Error! *****");
```

Comment:
yes, logger adds '\r\n' but there is an option to pass raw strings to it (log_printk() in log_core.h). Then when message is processed it is not formatted (no level prefix, no timestamp, no new line).

Reply:
Why is it adding both a \r and a \n?

Reply:
when printf is used console is implicitly adding \r when \n is detected. So log is adding both. If you look at sys_log, it was also adding \n and then through console adding \r.

Reply:
>  console is implicitly adding \r when \n is detected

You mean the UART console "backend"? (while it would seem the ipm backend actively removes the \r it finds, and the sys_log_net strips the trailing '\n'). Would it makes sense to have the logger backends be the one adding extra \r if it needs them and have the core of the logger be a passthru?

Reply:
It is like you said. Logger frontend/core doesn't do it. Log_output module is doing it. This helper module can be used by log backend to format string. We can extend log_ouput to optionally add \r.